canonical/maas-ui
# feat: Node action button group on MachineDetails MAASENG-1223

## Done

- Added group of contextual menus and buttons to replace "Take action" dropdown
- Implemented on machine details
- Hide group and replace with "Menu dropdown" on smaller screens

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the details page for a machine
- Ensure the action group renders correctly
- Click on each available action
- Ensure it opens the correct form
- Shrink your window horizontally
- Ensure the group is replaced by a "Menu" dropdown wrapping occurs within the button group
- Ensure that each available action in the menu opens the correct form

## Fixes

Fixes https://warthogs.atlassian.net/browse/MAASENG-1223

## Screenshots

### Before
![image](https://user-images.githubusercontent.com/35104482/213667607-288e1af7-1216-4911-bebb-2e415ec228b7.png)

### After
![image](https://user-images.githubusercontent.com/35104482/213728100-db2160e7-bba5-4adf-af02-faac30c083ae.png)

#### With lock switch
![image](https://user-images.githubusercontent.com/35104482/213728195-fa44647b-74a4-4f92-86b0-f4897d04d7fe.png)

#### Small screen
![image](https://user-images.githubusercontent.com/35104482/213728285-5312e547-abcc-47c2-9a94-f9bd4b215747.png)
